src/distributions/mod.rs

```diff
@@ -198,7 +205,7 @@ pub trait Distribution<T> {
 /// [`Distribution`]: trait.Distribution.html
 /// [`sample_iter`]: trait.Distribution.html#method.sample_iter
 #[derive(Debug)]
-pub struct DistIter<'a, D, R, T> where D: Distribution<T> + 'a, R: Rng + 'a {
+pub struct DistIter<'a, D: 'a, R: 'a, T> {
```

Comment:
I know some people advise making generic structs as generic as possible and only putting bounds on the implementations, but I can't see that we will ever want to use this without `D: Distribution<T>` or `R: Rng`; also imposing these constraints later would technically be a breaking change whereas relaxing them later wouldn't be. So I'd prefer to keep these constraints.

Reply:
I don't think it matters much. There is no reason to add our own trait bounds to structs, and I think we can safely assume that won't change.

But as @nagisa notes, if you would want to make a wrapper it would also require the same bounds, while this is not really necessary.